File: Assets/Typescript~/src/Shared/Entity/Character/CharacterEntity.ts
import { Inventory } from "Shared/Inventory/Inventory";
import { Entity, EntityDto } from "../Entity";
import { EntitySerializer } from "../EntitySerializer";

export interface CharacterEntityDto extends EntityDto {
	invId: number;
}

export class CharacterEntity extends Entity {
	private inventory: Inventory;

	constructor(id: number, networkObject: NetworkObject, clientId: number | undefined, inventory: Inventory) {
		super(id, networkObject, clientId);
		this.inventory = inventory;
	}

	public GetInventory(): Inventory {
		return this.inventory;
	}

	public Encode(): CharacterEntityDto {
		return {
			...super.Encode(),
			serializer: EntitySerializer.CHARACTER,
			invId: this.inventory.Id,
		};
	}
}

File: Assets/Bundles/Shared/Resources/TS/Entity/Character/CharacterEntity.lua
-- Compiled with unity-ts v2.1.0-75
local Entity = require("Shared/TS/Entity/Entity").Entity
local EntitySerializer = require("Shared/TS/Entity/EntitySerializer").EntitySerializer
local CharacterEntity
do
	local super = Entity
	CharacterEntity = setmetatable({}, {
		__tostring = function()
			return "CharacterEntity"
		end,
		__index = super,
	})
	CharacterEntity.__index = CharacterEntity
	function CharacterEntity.new(...)
		local self = setmetatable({}, CharacterEntity)
		return self:constructor(...) or self
	end
	function CharacterEntity:constructor(id, networkObject, clientId, inventory)
		super.constructor(self, id, networkObject, clientId)
		self.inventory = inventory
	end
	function CharacterEntity:GetInventory()
		return self.inventory
	end
	function CharacterEntity:Encode()
		local _object = {}
		for _k, _v in super.Encode(self) do
			_object[_k] = _v
		end
		_object.serializer = EntitySerializer.CHARACTER
		_object.invId = self.inventory.Id
		return _object
	end
end
return {
	CharacterEntity = CharacterEntity,
}
-- ----------------------------------
-- ----------------------------------
